server/tests/src/main/scala/sttp/tapir/server/tests/ServerSecurityTests.scala

```diff
@@ -108,6 +110,36 @@ class ServerSecurityTests[F[_], S, OPTIONS, ROUTE](createServerTest: CreateServe
         .serverLogic(s => _ => pureResult(s.asRight[Unit]))
     ) { (backend, baseUri) =>
       basicStringRequest.get(uri"$baseUri/auth").auth.bearer("1234").send(backend).map(_.body shouldBe "1234")
+    },
+    testServerLogic(
+      in_security_option_basic_out_string
+        .serverSecurityLogic((u: Option[UsernamePassword]) => pureResult(u.map(_.username).asRight[Unit]))
+        .serverLogic(s => _ => pureResult(s.toRight[Unit](()))),
+      "In security Option[UsernamePassword] should let in basic auth"
+    ) { (backend, baseUri) =>
+      basicStringRequest.get(uri"$baseUri/auth").auth.bearer("1234").auth.basic("a", "b").send(backend).map { a =>
```

Comment:
I think we should perform two requests: one with basic auth (only, why the bearer here?), and one without, checking the body that it matches the expected result